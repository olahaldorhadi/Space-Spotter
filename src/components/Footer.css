.footer--footer {
    background-color: var(--dark-blue);
    width: 100%;
    height: auto;
    display: flex;
    align-items: center;
    justify-content: center;
    padding-top: 10px;
    padding-bottom: 10px;
    bottom: 0;
    left: 0;
    right: 0;
}

.footer--footer p {
    color: var(--offwhite);
}

.footer--footer-content {
    max-width: 90%;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    margin-bottom: 1%;
}

.footer--logo {
    height: auto;
    width: 100%;
    max-width: 200px;
    margin-bottom: 10px;
}

.footer--footer-content img {
    height: 15vh;
    width: auto;
    max-width: 100%;
}

.footer--explanation-text {
    text-align: center;
    font-size: 1em;
    color: var(--offwhite);
    margin-top: 10px;
}

.footer--explanation-text h1,
.footer--explanation-text p {
    margin-top: 5px;
    margin-bottom: 10px;
}
