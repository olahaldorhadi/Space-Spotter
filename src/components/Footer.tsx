import './Footer.css';
import Logo from '../assets/SpaceSpotterLogo.png';

const Footer = () => {

  return (
    <>
      <div className='footer--footer'>
        <div className='footer--footer-content'>
          <div className='footer--explanation-text'>
          </div>
          <img src={Logo} />
        </div>
      </div>
    </>
  )
}

export default Footer
