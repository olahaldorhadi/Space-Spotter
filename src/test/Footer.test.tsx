import { render, screen } from '@testing-library/react'
import Footer from '../components/Footer'
import '@testing-library/jest-dom' // Import jest-dom to extend expect

test('renders Footer component', () => {
    render(<Footer />)

    // Check if the explanation text is present
    const explanationTextElement = screen.getByText(
        'Pssst, du kan trykke på et rom for å se mer informasjon eller opprette en favoritt'
    )
    expect(explanationTextElement).toBeInTheDocument()

    expect(screen.getByTestId('footer-component')).toMatchSnapshot()
})
