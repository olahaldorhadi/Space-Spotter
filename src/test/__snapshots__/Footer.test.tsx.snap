// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Footer component 1`] = `
<div
  class="footer--footer"
  data-testid="footer-component"
>
  <div
    class="footer--footer-content"
  >
    <div
      class="footer--explanation-text"
    />
    <p>
      Pssst, du kan trykke på et rom for å se mer informasjon eller opprette en favoritt
    </p>
  </div>
</div>
`;
